test/test_core.jl

```diff
@@ -242,6 +242,20 @@ end
     @test l(obj) == 1
     @test set(obj, l, 6) == (6,2,3)
 
+    obj = 123
+    for l in (@optic(_[1]), @optic(_[end]), @optic(_[]))
+        @test l(obj) === 123
+        @test set(obj, l, 456) === 456
+    end
+
+    obj = [:a, :b, :c]
+    l = @optic _[1:2]
+    @test @inferred(l(obj))::Vector{Symbol} == [:a, :b]
+    @test @inferred(set(obj, l, [:x, :y]))::Vector{Symbol} == [:x, :y, :c]
+    l = @optic _[CartesianIndex(2)]
+    @test @inferred(l(obj)) == :b
+    @test @inferred(set(obj, l, :x))::Vector{Symbol} == [:a, :x, :c]
+
```

Comment:
Here are some more tests I have used for the slice issue:

```julia
@assert setindex(rand(2,2), zeros(2), 1, :) isa Matrix{Float64}
@assert setindex(Array{Int, 0}(undef), "sdf") isa Array{Any, 0}
@assert setindex([rand(2) for i in 1:2, j in 1:2], zeros(2), 1, 1) isa Matrix{Vector{Float64}}
@assert setindex([rand(2) for i in 1:2, j in 1:2], [:bla], 1, 1) isa Matrix{Vector}
@assert setindex([rand(2) for i in 1:2, j in 1:2], :bla, 1, 1) isa Matrix{Any}
@assert setindex([rand(2) for i in 1:2, j in 1:2], [im, im], 1, :) isa Matrix{Any}
```

Reply:
I just dumped them here in case they're useful :D But there are no tests for slicing, as far as I can see, so maybe do include the first and last one?